engine/sound/src/devices/device_wasapi.cpp

```diff
@@ -332,13 +331,19 @@ namespace dmDeviceWasapi
             return dmSound::RESULT_OUT_OF_BUFFERS;
         }
 
+        const int channels_in = 2;
+        const int channels_out = device->m_MixFormat->nChannels;
         for (int i = 0; i < frames_available; ++i)
         {
             if (device->m_Format == WAVE_FORMAT_IEEE_FLOAT)
             {
                 float* fout = (float*)out;
-                fout[i*2+0] = samples[i*2+0]/32768.0f;
-                fout[i*2+1] = samples[i*2+1]/32768.0f;
+                fout[i*channels_out+0] = samples[i*channels_in+0]/32768.0f;
+                fout[i*channels_out+1] = samples[i*channels_in+1]/32768.0f;
+                for (int c = channels_in; c < channels_out; ++c)
+                {
+                    fout[i*channels_out+c] = 0;
+                }

```

Comment:
In this backend, we rely on the preferred settings the backend gives us. In this case, for a surround system, it return 8 channels (7.1), and we needed to write it accordingly, or we produced a lot of garbage.